src/bin/pgcopydb/pgcmd.h

```diff
@@ -182,6 +183,56 @@ typedef struct ArchiveContentArray
 	ArchiveContentItem *array;  /* malloc'ed area */
 } ArchiveContentArray;
 
+/* specify section of a dump: pre-data, post-data, data, schema */
+typedef enum
+{
+	PG_DUMP_SECTION_ALL = 0,
+	PG_DUMP_SECTION_SCHEMA,
+	PG_DUMP_SECTION_PRE_DATA,
+	PG_DUMP_SECTION_POST_DATA,
+	PG_DUMP_SECTION_DATA,
+	PG_DUMP_SECTION_ROLES       /* pg_dumpall --roles-only */
+} PostgresDumpSection;
+
+/*
+ * Convert PostgresDumpSection to string.
+ */
+static inline const char *
+postgresDumpSectionToString(PostgresDumpSection section)
+{
+	switch (section)
+	{
+		case PG_DUMP_SECTION_PRE_DATA:
+		{
+			return "pre-data";
+		}
+
+		case PG_DUMP_SECTION_POST_DATA:
+		{
+			return "post-data";
+		}
+
+		case PG_DUMP_SECTION_DATA:
+		{
+			return "data";
+		}
+
+		case PG_DUMP_SECTION_SCHEMA | PG_DUMP_SECTION_ROLES | PG_DUMP_SECTION_ALL:
```

Comment:
That's not the typical C spelling... this needs to look like other switch statements in the code base:

``` c
    switch (value)
    {
        case 1:
        case 2:
        {
            ...
        }
    }
```

Reply:
ah this is Rust syntax, good catch! 😄